adapters/socketio_berkeley.c

```diff
@@ -536,32 +536,31 @@ static int initiate_socket_connection(SOCKET_IO_INSTANCE* socket_io_instance)
         }
     }
 
-    if (socket_io_instance->address_type == ADDRESS_TYPE_IP)
-    {
-        socket_io_instance->socket = socket(addr->ai_family, addr->ai_socktype, addr->ai_protocol);
-    }
-    else
-    {
-        socket_io_instance->socket = socket(AF_UNIX, SOCK_STREAM, 0);
-    }
-
-    if (socket_io_instance->socket < SOCKET_SUCCESS)
-    {
-        LogError("Failure: socket create failure %d.", socket_io_instance->socket);
-        result = MU_FAILURE;
-    }
-#ifndef __APPLE__
-    else if (socket_io_instance->target_mac_address != NULL &&
-                set_target_network_interface(socket_io_instance->socket, socket_io_instance->target_mac_address) != 0)
-    {
-        LogError("Failure: failed selecting target network interface (MACADDR=%s).", socket_io_instance->target_mac_address);
-        result = MU_FAILURE;
-    }
-#endif //__APPLE__
-
     if(result == 0)
     {
-        if ((-1 == (flags = fcntl(socket_io_instance->socket, F_GETFL, 0))) ||
+        if (socket_io_instance->address_type == ADDRESS_TYPE_IP)
+        {
+            socket_io_instance->socket = socket(addr->ai_family, addr->ai_socktype, addr->ai_protocol);
+        }
+        else
+        {
+            socket_io_instance->socket = socket(AF_UNIX, SOCK_STREAM, 0);
+        }
+
+        if (socket_io_instance->socket < SOCKET_SUCCESS)
+        {
+            LogError("Failure: socket create failure %d.", socket_io_instance->socket);
+            result = MU_FAILURE;
+        }
+        #ifndef __APPLE__
+        else if (socket_io_instance->target_mac_address != NULL &&
+                    set_target_network_interface(socket_io_instance->socket, socket_io_instance->target_mac_address) != 0)
+        {
+            LogError("Failure: failed selecting target network interface (MACADDR=%s).", socket_io_instance->target_mac_address);
+            result = MU_FAILURE;
+        }
+        #endif //__APPLE__
+        else if ((-1 == (flags = fcntl(socket_io_instance->socket, F_GETFL, 0))) ||
             (fcntl(socket_io_instance->socket, F_SETFL, flags | O_NONBLOCK) == -1))
         {
             LogError("Failure: fcntl failure.");
```

Comment:
Code in line 580 is a bit unclear as well: it would help if we add a comment:


    if ((result != 0) && (errno != EINPROGRESS))
            {
                LogError("Failure: connect failure %d.", errno);
                result = MU_FAILURE;
            }
            else  // result == 0 || errno == EINPROGRESS
            {
                // Async connect will return -1.
                result = 0;                   // <----------- Note that result could already be 0 here.
                if (socket_io_instance->on_io_open_complete != NULL)
                {
                    socket_io_instance->on_io_open_complete(socket_io_instance->on_io_open_complete_context, IO_OPEN_OK /*: IO_OPEN_ERROR*/);
                }
            }